src/evilHangman/evilHangmanGame.ts

```diff
@@ -0,0 +1,163 @@
+import fs from 'fs';
+import { isNonEmptyArray } from '../helpers/guards/isNonEmptyArray';
+
+const DICTIONARY_PATH = './res/dictionary.txt';
+const allWords = fs
+	.readFileSync(DICTIONARY_PATH)
+	.toString()
+	.split('\n')
+	.map(word => word.trim());
+export class EvilHangmanGame {
+	private possibleWords: Array<string>;
+	private word: string;
+	private guessesRemaining: number;
+	private readonly guessesSoFar: Set<string>;
+
+	constructor(word: string, guessesRemaining: number, guessesSoFar: Set<string>) {
+		this.word = word;
+		this.guessesRemaining = guessesRemaining;
+		this.guessesSoFar = guessesSoFar;
+
+		this.possibleWords = allWords.filter(possibleWord => this.word.length === possibleWord.length);
+		this.removePossibleWords([...guessesSoFar]);
+	}
+
+	static newGame(length: number | null, guesses: number | null): EvilHangmanGame {
+		if (length === null) {
+			length = allWords[Math.floor(Math.random() * allWords.length)]?.length ?? 1;
```

Comment:
I get that this line returns a random word in the dictionary, but only because I've used this pattern before. Can we create a helper function to make this part clearer, or at least write a comment explaining this thing?

Reply:
done